IntegrationWithPharmacies/Controllers/ReportController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using HealthClinic.CL.DbContextModel;
+using HealthClinic.CL.Model.Orders;
+using HealthClinic.CL.Repository;
+using HealthClinic.CL.Service;
+using IntegrationWithPharmacies.FileProtocol;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace IntegrationWithPharmacies.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ReportController : ControllerBase
+    {
+        private ReportService ReportService { get; set; }
+        private DoctorOrderServica DoctorOrderService { get; set; }
+        private MedicineForOrderingService MedicineService { get; set; }
+
+        public ReportController(MyDbContext context)
+        {
+            ReportService = new ReportService();
+            MedicineService = new MedicineForOrderingService(context);
+            DoctorOrderService = new DoctorOrderServica(context);
+        }
+ 
+
+        [HttpPost] 
+        public IActionResult Post(DateOfOrder date)
+        {
+            var sftpService = new SftpService(new NullLogger<SftpService>(), getConfig());
+            var testFile = @"..\TextFile.txt";
+            System.IO.File.WriteAllText(testFile,getReportText(date));
+            sftpService.UploadFile(testFile, @"\pub\" + Path.GetFileName(testFile));
+            return Ok();
+        }
+
+        public String getReportText(DateOfOrder date)
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+            int totalQuatity = 0;
+            int i = 1;
+            stringBuilder.Append("Report about consumption of medicine\n\n\n");
+            foreach (DoctorsOrder order in DoctorOrderService.GetAll())
+            {
+                foreach (MedicineForOrdering medicine in MedicineService.GetAll())
+                {
+                    if (medicine.OrderId.Equals(order.id) && order.IsFinished && DateTime.Compare(order.DateEnd, convertStringToDate(date.StartDate)) == 1 && DateTime.Compare(order.DateEnd, convertStringToDate(date.EndDate)) == -1)
+                    {
+                        stringBuilder.Append(i + ".\n     Medicine name: " + medicine.Name + "\n     Ordered quantity: " + medicine.Quantity + " (Date:  " + order.DateEnd.Date.ToString() + ")\n");
+                        totalQuatity += medicine.Quantity;
+                        i++;
+                    }
+                }
+            }

```

Comment:
Refactor the code and fix clean code mistakes. 
One option is to make separate methods to avoid two foreach loops.